cognite/extractorutils/rest/http.py

```diff
@@ -1,13 +1,24 @@
 from enum import Enum
-from typing import Generic, TypeVar
+from typing import Callable, Dict, Generic, List, TypeVar, Union
 from urllib.parse import urlparse
 from uuid import UUID
 
 import arrow
 
-RequestBody = TypeVar("RequestBody")
 ResponseType = TypeVar("ResponseType")
 
+# Ignoring types here, since recursive types are not yet supported by mypy:  https://github.com/python/mypy/issues/731
```

Comment:
We could investigate other type checkers until `mypy` fixes this. I believe `pyright` already supports this.